chain/generate_address_from_pubkey.js

```diff
@@ -0,0 +1,38 @@
+const bjs = require("bitcoinjs-lib");
+const ecc = require("tiny-secp256k1");
+const { BIP32Factory } = require("bip32");
+const b58 = require("bs58check");
+const bip32 = BIP32Factory(ecc);
+
+module.exports = ({ network, masterPubKey, index = 0 }) => {
+  //set the network
+  if (network === "mainnet") {
+    network = bjs.networks.bitcoin;
+  }
+  if (network === "testnet") {
+    network = bjs.networks.testnet;
+  }
+  if (network === "regtest") {
+    network = bjs.networks.regtest;
+  }
+
+  const xpub = anypubToXpub(masterPubKey);
+
+  //generate an address, using p2sh wrapper over p2wpkh to support legacy wallets
```

Comment:
is there some way to tell which type of master public key it is and then generate the correct address type for that key?

Reply:
ok so if the xpub can't tell you what type of address it wants, how about having that as a parameter so you could have native segwit or nested segwit